lesson6/main.js

```diff
@@ -0,0 +1,33 @@
+const ul = document.getElementById('js-ul');
+const frag = document.createDocumentFragment();
+
+const lists = [
+  {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, 
+  {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+]
+
+function getListsObj() {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(lists);
+    },3000);
+  });
+}
+
+getListsObj().then((lists)=> {
+  lists.forEach(list => {
+    const li = document.createElement('li');
+    const anchor = document.createElement('a');
+    const img = document.createElement('img');
+
+    anchor.href = `/${list.to}`;
+    img.src = list.img;
+    img.alt = list.alt;
+
+    frag.appendChild(li).appendChild(anchor).appendChild(img);
+    anchor.insertAdjacentHTML('beforeend',list.text);
+  });
+  ul.appendChild(frag);
+});
```

Comment:
It reads and executes the `lists` in "then" without using the arguments passed in `resolve`.

> Promiseオブジェクトを使って解決された値として受け取り

The problem says this, so try to use the arguments that have been passed across.

Reply:
I think the problem with the naming is that the `lists` declared as an array and the `lists` passed as resolved values are the same.

Reply:
Thanks for the review!
As a result of using the same name as the object, we were not able to receive the value...
I have fixed it.
Please check.
83612a9fee2a32a9e8842a575fc569d10d470446